src/rapidhtml/tags.py

```diff
@@ -126,6 +142,99 @@ def render(self):
+
+        Raises:
+            KeyError: If no matching tags are found and pop is True.
+            ValueError: If an instantiated object is passed as the tag parameter.
+            TypeError: If the tag parameter is of an unexpected type.
+
+        """
+        if isinstance(select_tag, BaseTag):
+            raise ValueError("Cannot pass instantiated object to select method!")
+        elif inspect.isclass(select_tag) and issubclass(select_tag, BaseTag):
+            select_tag_name = select_tag.__name__.lower()
+        elif isinstance(select_tag, str):
+            select_tag_name = select_tag
+        else:
+            raise TypeError(f"Unexpected tag type {type(select_tag)}. {select_tag}")
+
+        ret_tags: list["BaseTag"] = []
+        self_matches: list[tuple[int, "BaseTag"]] = []
+
+        for i, tag in enumerate(self.tags):
+            if not isinstance(tag, BaseTag):
+                continue
+            tag_name = tag.tag_name
+            if tag_name == select_tag_name or tag_name == f"htmltag{select_tag_name}":
+                self_matches.append((i, tag))
+                if first_match:
+                    break
+            elif recurse and isinstance(tag, BaseTag):
+                sub_ret_tags = tag.select(
+                    select_tag, recurse=recurse, pop=pop, first_match=first_match
+                )
+                if sub_ret_tags:
+                    ret_tags.extend(sub_ret_tags)
+                    if first_match:
+                        break
+
+        if not (ret_tags or self_matches) and pop:
+            raise KeyError(f"Unable to find tag {select_tag}")
+
+        # Pop in reverse order to avoid IndexError!
+        for i, tag in sorted(self_matches, reverse=True, key=lambda tpl: tpl[0]):
+            if pop:
+                self.tags.pop(i)
+            ret_tags.append(tag)
+
+        return ret_tags
+
+    def pop(self, tag: typing.Type["BaseTag"] | str, *default: T) -> "BaseTag" | T:
+        """
+        Selects and removes the first occurrence of a tag that matches the given tag name or BaseTag object.
+        If no matching tag is found, it returns the default value or raises a KeyError if no default is provided.
+
+        Args:
+            tag (Type[BaseTag] or str): The uninstantiated subclass of BaseTag or the tag name to match against.
+            *default (Optional[T]): Optional default value to return if no matching tag is found. Defaults to None.
+
+        Returns:
+            BaseTag or T: The selected tag or the default value if no matching tag is found.
+
+        Raises:
+            KeyError: If no matching tags are found and no default value is provided.
+            ValueError: If an instantiated object is passed as the tag parameter.
+            TypeError: If the tag parameter is of an unexpected type.
+
+        """
+        if default and len(default) > 1:
+            raise TypeError("pop expected at most 2 arguments, got more")
```

Comment:
I'm missing the reason why we're even allowing more than one value...

Reply:
Ahhhhhhhh! That's weird but I like it! I never even thought about how that actually functioned. Well thought out

Reply:
But when using `*args`, but cannot restrict the user to how many params they pass, but check the values on the other end .. which is exactly what dict.pop does as well!

```bash
$ python -c '{"a": 1}.pop("b", None, None, None, None)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: pop expected at most 2 arguments, got 5

```